version: '3.8'
services:
  zookeeper:
    image: confluentinc/cp-zookeeper:7.4.0
    environment:
      ZOOKEEPER_CLIENT_PORT: 2181
    ports:
      - "2181:2181"
    networks:
      - app_app_network

  kafka:
    image: confluentinc/cp-kafka:7.4.0
    depends_on:
      - zookeeper
    ports:
      - "9092:9092"
    environment:
      KAFKA_BROKER_ID: 1
      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://kafka:9092
      KAFKA_LISTENERS: PLAINTEXT://0.0.0.0:9092
      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
    healthcheck:
      test: ["CMD", "nc", "-z", "localhost", "9092"]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - app_app_network

  postgres:
    image: postgres:15
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: data_processor_db
    ports:
      - "5432:5432"
    networks:
      - app_app_network

  data-processor:
    build:
      context: ./services/app-data-processor
    env_file:
      - ./env/.env
    environment:
      - PYTHONPATH=/app/app
    depends_on:
      kafka:
        condition: service_healthy
      postgres:
        condition: service_healthy
    volumes:
      - ./services/data_processor/logs:/app/logs
    restart: unless-stopped
    networks:
      - app_app_network

  tg_standard_bot:
    build:
      context: ./services/tg_standard_bot
    env_file:
      - ./env/.env
    depends_on:
      kafka:
        condition: service_healthy
    volumes:
      - ./services/tg_standard_bot/app/logs:/app/logs
    restart: no
    networks:
      - app_app_network

  tg_user_bot:
    build:
      context: ./services/tg_user_bot
    env_file:
      - ./env/.env
    environment:
      - LOG_DIR=/app/logs
    depends_on:
      kafka:
        condition: service_healthy
      postgres:
        condition: service_healthy
    volumes:
      - ./services/tg_user_bot/logs:/app/logs
      - ./services/tg_user_bot/media:/app/media
      - ./session_name.session:/app/session_name.session
    restart: no
    networks:
      - app_app_network

networks:
  app_app_network:
    name: app_app_network
    driver: bridge
